core/src/tx.rs

```diff
@@ -128,6 +125,50 @@ impl TransactionValidator {
             }));
         }
 
+        // WSV is cloned here so that instructions don't get applied to the blockchain
+        // Therefore, this instruction execution validates before actually executing
+        let wsv_for_builtin_validators = WorldStateView::clone(&self.wsv);
+        self.validate_with_builtin_validators(&tx, &wsv_for_builtin_validators, is_genesis)?;
+
+        // Making a new clone so that instructions applied in the previous step won't break validation
```

Comment:
It's already a significant performance hit, because we are already clonning wsv for checking. I'm just doing it once more.
In future we will remove builtin validators and we will have one clonning as before

Reply:
Uh, seems like we haven't any workaround for now.
> I'm just doing it once more.

If validation is in the hot path it could significant effect, however without bench-marking it's just speculations.
Could you try to run `tps` benchmark with and without addition cloning?

Reply:
After running on `iroha2-dev` and then on my branch benchmark reported that **no changes detected**